src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -242,26 +255,83 @@ void AddReceivers(IBuilder builder, RecoverabilityExecutorFactory recoverability
                 var instanceSpecificRecoverabilityExecutor = recoverabilityExecutorFactory.CreateDefault(instanceSpecificQueue);
                 var sharedReceiverPushSettings = new PushSettings(instanceSpecificQueue, errorQueue, configuration.PurgeOnStartup, requiredTransactionSupport);
 
-                receivers.Add(new TransportReceiver(MainReceiverId, messagePumpFactory(), sharedReceiverPushSettings, dequeueLimitations, mainPipelineExecutor, instanceSpecificRecoverabilityExecutor, criticalError));
+                receivers.Add(new TransportReceiver(MainReceiverId, messagePumpFactory, sharedReceiverPushSettings, dequeueLimitations, mainPipelineExecutor, instanceSpecificRecoverabilityExecutor, criticalError, systemOutageConfiguration.RateLimitStartedNotification, systemOutageConfiguration.RateLimitEndedNotification));
             }
 
             foreach (var satellitePipeline in configuration.SatelliteDefinitions)
             {
                 var satelliteRecoverabilityExecutor = recoverabilityExecutorFactory.Create(satellitePipeline.RecoverabilityPolicy, satellitePipeline.ReceiveAddress);
                 var satellitePushSettings = new PushSettings(satellitePipeline.ReceiveAddress, errorQueue, configuration.PurgeOnStartup, satellitePipeline.RequiredTransportTransactionMode);
 
-                receivers.Add(new TransportReceiver(satellitePipeline.Name, messagePumpFactory(), satellitePushSettings, satellitePipeline.RuntimeSettings, new SatellitePipelineExecutor(builder, satellitePipeline), satelliteRecoverabilityExecutor, criticalError));
+                receivers.Add(new TransportReceiver(satellitePipeline.Name, messagePumpFactory, satellitePushSettings, satellitePipeline.RuntimeSettings, new SatellitePipelineExecutor(builder, satellitePipeline), satelliteRecoverabilityExecutor, criticalError, systemOutageConfiguration.RateLimitStartedNotification, systemOutageConfiguration.RateLimitEndedNotification));
             }
         }
 
+        async Task RateLimitLoop(CancellationToken cancellationToken)
+        {
+            //We want all the pumps to be running all the time in the desired mode until we call stop
+            //We want to make sure that StopRateLimit signal is not lost
+            //The circuit breaker ensures that if StartRateLimit has been called that eventually StopRateLimit is going to be called unless the endpoint stop
+
+            while (true)
+            {
+                await transitionBetweenRateLimitModes.WaitAsync(cancellationToken).ConfigureAwait(false);
+                var startRateLimiting = endpointShouldBeRateLimited;
+
+                while (true)
+                {
+                    try
+                    {
+                        foreach (var receiver in receivers)
+                        {
+                            if (startRateLimiting)
+                            {
+                                await receiver.StartRateLimiting().ConfigureAwait(false);
```

Comment:
Why not change the design by simply calling Start and Stop and bringing in the PushRuntimeSettings via the Start then rate limiting wouldn't have to leak into the receiver